pkg/controller/instance/instance_controller.go

```diff
@@ -222,6 +247,17 @@ func updateInstance(instance *kudov1beta1.Instance, oldInstance *kudov1beta1.Ins
 		log.Printf("InstanceController: Error when updating instance status. %v", err)
 		return err
 	}
+
+	// update instance metadata if finalizer is removed
+	// because Kubernetes will immediately delete the instance, this has to be the last instance update
+	maybeRemoveFinalizer(instance)
```

Comment:
this is still 2 update calls, do we have to do that? this is first update https://github.com/kudobuilder/kudo/pull/1108/files#diff-45e4638e2c718ccd5d08635baec770b1R236, can't we just remove the finalizer from the instance object before the first update and do that all on one request?

Reply:
If we would do that, `client.Status().Update()` will fail, because once the finalizer has been removed, the instance will be deleted (i.e. the update acts like a delete). Though we could add some code to handle this case and _not_ call `client.Status().Update()` if the `client.Update()` removed the finalizer. But this seems to complicate the code a lot, hence I chose the simpler approach, though being aware that it results in 2 update calls.

Reply:
oh I get it now, thanks!

Reply:
Though, this can be avoided if we run `client.Status().Update()` first. Looks like ordering isn't important here so this change is possible.

Reply:
Hmm, some tests are failing after changing the ordering. Looks like it _is_ important.